CVAE_testbed/datasets/quilt_aics_features.py

```diff
@@ -0,0 +1,239 @@
+
+            self._color = X_train[:, 0]
+        
+            C_train = X_train.clone()
+            C_test = X_test.clone()
+
+            count = 0
+            if self.shuffle is True:
+                while count == 0:
+                    C_mask_train = torch.zeros(C_train.shape).bernoulli_(0.5)
+                    C_mask_test = torch.zeros(C_test.shape).bernoulli_(0.5)
+                    count=1 
+            else:
+                C_mask_train = torch.zeros(C_train.shape).bernoulli_(0)
+                C_mask_test = torch.zeros(C_test.shape).bernoulli_(0)
+
+            C_train[C_mask_train.byte()] = 0
+            C_train_indicator = C_mask_train == 0
+
+            C_test[C_mask_test.byte()] = 0
+            C_test_indicator = C_mask_test == 0
+            print('Out')
+
+            C_train = torch.cat([C_train.float(), C_train_indicator.float()], 1)
+            C_test = torch.cat([C_test.float(), C_test_indicator.float()], 1)
+
+            X_train = X_train.view([1, -1, X_train.size()[-1]])
+            X_test = X_test.view([1, -1, X_test.size()[-1]])
+            C_train = C_train.view([1, -1, X_train.size()[-1]*2])
+            C_test = C_test.view([1, -1, X_test.size()[-1]*2])
+
+            # Sum up
+            conds_train = C_train[:, :, X_train.size()[-1]:].sum(2)
+            conds_test = C_test[:, :, X_test.size()[-1]:].sum(2)
+
+            Batches_X_train = torch.cat([Batches_X_train, X_train], 0)
+            Batches_C_train = torch.cat([Batches_C_train, C_train], 0)
+            Batches_conds_train = torch.cat([Batches_conds_train, conds_train], 0)
+            try:
+                Batches_X_test = torch.cat([Batches_X_test, X_test], 0)
+                Batches_C_test = torch.cat([Batches_C_test, C_test], 0)
+                Batches_conds_test = torch.cat([Batches_conds_test, conds_test], 0)
+            except:
+                pass
+
+        self._batches_x_train = Batches_X_train
+        self._batches_c_train = Batches_C_train
+        self._batches_conds_train = Batches_conds_train
+
+        self._batches_x_test = Batches_X_test
+        self._batches_c_test = Batches_C_test
+        self._batches_conds_test = Batches_conds_test
+    
+    def __len__(self):
+        return len(self._batches_x)
+
+    def __getitem__(self, idx):
+        """
+        Returns a tuple. (X, C, sum(C[mid:end])). 
+        X is the input, 
+        C is the condition, 
+        sum(C[mid:end]) is the sum of the indicators in C. It tells us how many of the condition
+        columns have been masked
+        """
+        return self._batches_x[idx], self._batches_c[idx], self._batches_conds[idx]
+
+    def get_train_data(self):
+        return self._batches_x_train, self._batches_c_train, self._batches_conds_train
+
+    def get_test_data(self):
+        return self._batches_x_test, self._batches_c_test, self._batches_conds_test
+    
+    def __len__(self):
+        return len(self._batches_x)
+
+    def __getitem__(self, idx):
```

Comment:
Wait a second... you have two `__getitem__` functions? How does Python not complain about this...

Reply:
huh